pkg/apis/topology/v1alpha1/types.go

```diff
@@ -29,24 +29,32 @@ type NodeResourceTopology struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	TopologyPolicy string        `json:"topologyPolicy"`
-	Nodes   []NUMANodeResource   `json:"nodes"`
+	TopologyPolicies []string        `json:"topologyPolicies"`
+	Zones            map[string]Zone `json:"zones"`
 }
 
-// NUMANodeResource is the spec for a NodeResourceTopology resource
-type NUMANodeResource struct {
-	NUMAID int
-	Resources v1.ResourceList
+// Zone is the spec for a NodeResourceTopology resource
+type Zone struct {
+	Type       string
```

Comment:
No json tags here for keys?
```
type Zone struct {
 Type string  `json:"type"`
 Parent string  `json:"parent"`
 Cost map[string]int  `json:"cost"`
 Attributes map[string]string  `json:"attributes"
 Resources  ResourceInfoMap  `json:"resources"
}
```

Reply:
As I remember if json is omitted, json field name  is given as golang field name, but we need to specify is that field mandatory or not, so yes. I'll add json. Thanks.